presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java

```diff
@@ -257,19 +278,16 @@ private synchronized void initiateRequest()
                         .setUri(uri).build(),
                 new PageResponseHandler(blockEncodingSerde));
 
-        Futures.addCallback(future, new FutureCallback<PagesResponse>()
+        HttpResponseFuture<PagesResponse> resultFuture = (HttpResponseFuture<PagesResponse>) future;
```

Comment:
I would do this the other way around.  Assign `executeAsync` result directly to the variable `resultFuture` and then in the second statement `this.future = resultFuture`. That way you don't need the cast.